src/sentry/monitors/endpoints/monitor_ingest_checkin_attachment.py

```diff
@@ -132,7 +137,13 @@ def convert_args(
 
         # When looking up via GUID we do not check the organization slug,
         # validate that the slug matches the org of the monitors project
-        if organization_slug and project.organization.slug != organization_slug:
+        if not organization_slug:
+            raise ResourceDoesNotExist
+
+        if options.get("api.id-or-slug-enabled") and str(organization_slug).isnumeric():
+            if project.organization.id != organization.id:
+                raise ResourceDoesNotExist
+        elif project.organization.slug != organization_slug:
```

Comment:
I think this logic is a bit different than what was originally there. We only raise if the `organization_slug` was set and it doesn't match.

This is a bit implicit b/c we're not checking the option, but rather relying on the fact that both slug and id are unique. Prob good to add a comment explaining this?
```suggestion
        if organization_slug and project.organization.slug != organization_slug and project.organization.id != organization_slug:
                raise ResourceDoesNotExist
```

Reply:
good catch! 🥔